MdsSupportingClasses/Collivery.php

```diff
@@ -851,29 +866,35 @@ public function addAddress(array $data)
             $this->setError('missing_data', 'street not set.');
         }
 
-        if (!isset($data['full_name'])) {
+        if (!isset($data['contact']['full_name'])) {
             $this->setError('missing_data', 'full_name not set.');
         }
 
-        if (!isset($data['phone']) and !isset($data['cellphone'])) {
+        if (!isset($data['contact']['work_phone']) and !isset($data['contact']['cellphone'])) {
             $this->setError('missing_data', 'Please supply ether a phone or cellphone number...');
         }
 
+        if (isset($data["custom_id"]) && is_integer($data["custom_id"])) {
+            if ($data["custom_id"] == 0) {
+                $data["custom_id"] = null;
+            }
+        }
+
         if (!$this->hasErrors()) {
             try {
-                $result = $this->client()->add_address($data, $this->token);
+                $result = $this->consumeAPI("https://api.collivery.co.za/v3/address", $data, 'POST');
                 $this->cache->forget('collivery.addresses.'.$this->client_id);
             } catch (SoapFault $e) {
                 $this->catchSoapFault($e);
 
                 return false;
             }
 
-            if (isset($result['address_id'])) {
-                return $result;
+            if (isset($result['data']['id'])) {
```

Comment:
The following code block is repeated several times:
```php
if (isset($result['data'])) {
 // Do something
} else {
    if (isset($result['error'])) {
        $this->setError($result['error']['http_code'], $result['error']['message']);
    } else {
        $this->setError('result_unexpected', 'No result returned.');
    }

    return false;
}
```

I think this code might be better inside of the `consumeAPI()` method.
All of the repetitions can be removed.

Reply:
So I can't move it into the consumeAPI() method, as the initial if condition isn't the same and the content happening inside the if is different among them all. 

I've moved the else section into a function as that is repeated all over.